### ISSUE TITLE
How to add custom drivers non-intrusively?

### ISSUE THREAD

**Issue by FlyingSamson** [ess-dmsc/h5cpp#650]
For the project I'm currently working on, I need to write a custom driver.

Inheriting from `hdf5::file::Driver` works per se, but to overload the pure virtual function `id()` I need to return a `hdf5::file::DriverID`, which currently is limited to `Posix`, `Direct`, `Memory`, and `MPI`. 

I could not find any reference to `hdf5::file::DriverID` other than the once in the existing drivers `id()` functions, so I'm not sure what the intended use `id()` is.

Would it be feasible to add another choice (e.g., `Custom`) to the `hdf5::file::DriverID` enum?

Alternatively, if the ID should remain unique even when multiple custom drivers are to be used, one might also consider some form of registration, that hands out unique integer ids for each driver. In any case I believe driver should be implemented as an open set, since hdf5 itself implements it that way. The implementation using the enum class makes this difficult, as it inherently makes it a closed set.

**Comment by jkotan** [comment]
Yes, it would be nice to support `H5Pset_driver` and `H5Pget_driver` . I would think about a solution which is backward compatible

**Comment by jkotan** [comment]
Hello @FlyingSamson, 

I've just created PR which adds DriverID::Custom=0 (https://github.com/ess-dmsc/h5cpp/pull/665)
Does it solve your issue? 
Do you have any simple example for your use-case we could add to our tests?
